op-node/node/safedb/safedb.go

```diff
@@ -0,0 +1,112 @@
+package safedb
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"math"
+	"sync/atomic"
+
+	"github.com/cockroachdb/pebble"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+var (
+	ErrNotFound     = errors.New("not found")
+	ErrInvalidEntry = errors.New("invalid db entry")
+)
+
+type SafeDB struct {
+	log log.Logger
+	db  *pebble.DB
+
+	writeOpts *pebble.WriteOptions
+
+	closed atomic.Bool
+}
+
+func KeyL1BlockNum(num uint64) []byte {
+	key := make([]byte, 0, 9)
+	key = append(key, 0)
+	key = binary.LittleEndian.AppendUint64(key, num)
```

Comment:
This should be big-endian if we want key-iteration to be sequential in block-numbers, I think. It likely depends on pebble key iteration, but that would make sense to me.